packages/analytics/amplify_analytics_pinpoint/example/integration_test/auto_session_tracking_test.dart

```diff
@@ -0,0 +1,103 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// A copy of the License is located at
+//
+//  http://aws.amazon.com/apache2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+import 'dart:async';
+
+import 'package:amplify_analytics_pinpoint_dart/amplify_analytics_pinpoint_dart.dart';
+import 'package:amplify_flutter/amplify_flutter.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+
+import 'utils/mock_lifecycle_provider.dart';
+import 'utils/setup_utils.dart';
+import 'utils/test_event.dart';
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  group('auto session tracking', () {
+    final mockLifecycleObserver = MockLifecycleProvider();
+
+    late Stream<TestEvent> eventsStream;
+
+    setUp(() async {
+      eventsStream = await configureAnalytics(
+        appLifecycleProvider: mockLifecycleObserver,
+      );
+    });
+
+    testWidgets(
```

Comment:
I think this test is more complicated than it needs to be, but maybe I am missing something. There seems to be some duplication in the test, and the use of streams might be complicating things more than needed.

Would this not test the same thing?

```dart
mockLifecycleObserver.triggerOnBackgroundListener();

final event1 = await eventsStream.first;

expect(event1.eventType, zSessionStopEventType);

mockLifecycleObserver.triggerOnForegroundListener();
await Amplify.Analytics.flushEvents(); // not sure if this is needed, but it is in current test

final event2 = await eventsStream.first;

expect(event2.eventType, zSessionStartEventType);

expect(event1.session.sessionId, isNot(event2.session.sessionId));

expect(
  event2.session.startTimestamp.isAfter(event1.session.stopTimestamp!),
  isTrue,
  reason: 'onForeground was called after onBackground',
);
```